### ISSUE TITLE
Create "Nightly" version of activity-streams add-on

### ISSUE THREAD

**Issue by pdehaan** [mozilla/activity-streams#442]
This was something @nchapman and I were casually talking about the other day, but I don't think we have a tracking issue for.

In some cases it may be nice if we had a version of the **activity-streams** add-on that either updated daily on a cron job, or maybe built+released on every commit.

We were thinking of 3 possible approaches:
- Stand up some Jenkins server to watch the repo and build the add-on (or build on a schedule), and then push to S3 or somewhere.
- Use Travis-CI and encrypted keys to do the add-on signing and push to S3 bucket.
- Stand up a simple server on AWS which listens for some GitHub webhook and does a "git pull" and then signs and builds the add-on.

This would be different from our regular "official" add-on which we distribute via TestPilot and is released every 1-2 weeks. This would be more for QA to rapidly test the latest features (without having to do Git+npm), and would let contributors play with the bleeding edge builds.

**Comment by oyiptong** [comment]
We were also talking about something like that, to have a dev release and a production release

**Comment by pdehaan** [comment]
We were thinking Travis-CI may be able to do all this for us already, as long as we encrypt our AMO keys and boto/S3 creds. We'd just need to run the [fabfile.py](https://github.com/mozilla/activity-streams/blob/master/fabfile.py) which does all the XPI packaging and S3 uploading.
Not sure if we'd just have to tweak the package.json version on Travis to be something like "1.0.5-2016-04-01-{git_sha}" to make it unique and let us know exactly what codebase we're using.

Using Jenkins seems like overkill, but may be easiest (at the expense of having a Jenkins instance to patch and maintain).

AWS sounds like the most fun and flexible, and we'd have some name like `https://activity-streams.dev.mozaws.net" which we could use to serve the XPI and update.rdf and keep it nicely firewalled from our S3 flow.

**Comment by pdehaan** [comment]
Actually, I think this may be super trivial w/ Travis-CI and using [my favorite] surge.sh: ["Integrating [Surge] with Travis CI"](https://surge.sh/help/integrating-with-travis-ci).

Looks like we'd just need to generate a Surge.sh auth token, create a couple of secure env vars (on Travis site), `npm install surge -g` (in .travis.yml), and then add an `after_success` Travis step to push our ./dist/ directory to some free *.surge.sh domain. No need for S3 or boto or other creds. We'd just need to add a couple secure AMO tokens so we can have Travis sign the addon (without exposing the creds).

```yaml
after_success:
  - surge --project ./dist --domain activity-streams-nightly.surge.sh
```

**Comment by pdehaan** [comment]
This was ridiculously easy w/ Travis-CI and deploying to surge.sh. You can check out my sample repo at https://github.com/pdehaan/surge_travis and look at my [Travis build logs](https://travis-ci.org/pdehaan/surge_travis/builds) to see if anything was leaked, such as my surge.sh email address or token.

Although I'm not a Travis-CI expert, so I'm not 100% sure if this will do _exactly_ what we want.
Right now it seems to be deploying to my dummy http://auto-deploy-test.surge.sh/ site on every PR. Not sure if there is an easy way to only do that on the code once a PR is merged into master. Otherwise I have a weird gut feeling that if we rebuild the XPI and deploy the latest.xpi and update.rdf on each commit, then we'll have features popping in and out everytime somebody pushes a new PR and overwrites the last version.

Plus, Travis is aware of Git commit SHAs, so it should be easy enough to read the package.json file, and append the `$TRAVIS_BRANCH` and/or `$TRAVIS_COMMIT` to the `version` before building the XPI. That way instead of a constant version like "1.0.9", we could do "1.0.9-master-27548d".

But there may be some simple Travis matrix tricks, or checking the `$TRAVIS_BRANCH` before deploying, or some other hack I'm not thinking of.

/cc @nchapman
